libs/libc/modlib/modlib_sections.c

```diff
@@ -66,6 +66,7 @@
  *
  ****************************************************************************/
 
+#if 0 /* Not used */
```

Comment:
How about we remove #if 0 from modlib_findsection? The compiler/linker will optimize out the unused function.

Reply:
i just followed the existing convention. i'm fine with either ways.
iirc, the linker optimization traditionally works at object level, not functions, though.

Reply:
@patacongo i think you added #if 0 on modlib_findsection. do you have any opinion?

Reply:
> i just followed the existing convention. i'm fine with either ways.
> iirc, the linker optimization traditionally works at object level, not functions, though.

Yes, we may need to enable gc, but if user take care memory consumption, he should eanble gc anyway.

Reply:
> i think you added #if 0 on modlib_findsection. do you have any opinion?

If it isn't used, why waste memory.

It seems to me that there are two global functions in that file that are not closely related.  I think they should pull out into separate files.  I would do this:

1) Move modlib_loadshdrs() into a new file.  I don't understand why it is in that file, and
2) Remove the #if 0

Then modlib_findsection() will be brought into build only if it is used with no dependency to lto.

Reply:
I don't see any reason why modlib_loadshdrs() and modlib_findsection() are in the same file.  Do you see some dependency?  They there is not some shared dependency or very close functional releationship, they belong in separate files.

Reply:
@patacongo a good idea. done.
i also replaced your copyright notices in this directory with Apache 2.0 as i added a new file with the same license.

Reply:
Thanks!  Pretty soon, my name won't be on any of the code.  I have mixed feelings about that after almost 25 years of working with this code (and daily for the last 15 years) with over 36 thousand commits.  But it is the best for the future.

Reply:
isn't "Nutt" of "NuttX" your name? i guess it will be there ~forever.